theorems/homotopy/EilenbergMacLane1.agda

```diff
@@ -72,31 +74,460 @@ module homotopy.EilenbergMacLane1 {i} (G : Group i) where
       transport (λ x → G.ident == x [ fst ∘ Codes ↓ emloop g ])
                  (G.unit-l g) (↓-Codes-loop g G.ident)
 
+
+    -- TODO: move this elsewhere
+    foo : ∀ {i j k} {A : Type i} {B : A → Type j} {C : A → Type k}
+      {x x' : A} {p₁ p₂ : x == x'} (q : p₁ == p₂)
+      (u : B x → C x) (u' : B x' → C x')
+      (s : transport C p₁ ∘ u == u' ∘ transport B p₁)
+      (t : transport C p₂ ∘ u == u' ∘ transport B p₂)
+      → (s ∙' ap (λ p → u' ∘ transport B p) q) == (ap (λ p → transport C p ∘ u) q ∙ t)
+      -- → s == t [ (λ p → transport C p ∘ u == u' ∘ transport B p) ↓ q ]
+      → ↓-→-from-transp s == ↓-→-from-transp t [ (λ p → u == u' [ (λ y → B y → C y) ↓ p ]) ↓ q ]
+    foo q u u' s t h = ap↓ ↓-→-from-transp (↓-='-in h)
+
+    module Decode where
+
+      P : EM₁ G → Type i
+      P x' = fst (Codes x') → embase == x'
+
+      Q : embase' G == embase → Type i
+      Q p = emloop == emloop [ P ↓ p ]
+
+      {- transport in path fibration of EM₁ -}
+      transp-PF : embase' G == embase → embase' G == embase → embase' G == embase
+      transp-PF = transport (λ x → embase' G == x)
+
+      {- transport in Codes -}
+      transp-C : embase' G == embase → G.El → G.El
+      transp-C = transport (fst ∘ Codes)
+
+      transp-emloop : embase' G == embase → G.El → embase' G == embase
+      transp-emloop p = transp-PF p ∘ emloop
+
+      emloop-transp : embase' G == embase → G.El → embase' G == embase
+      emloop-transp p = emloop ∘ transp-C p
+
+      module _ (g : G.El) where
+
+        module _ (y : G.El) where
+
+          transp-emloop=∙-emloop : transp-emloop (emloop g) y == emloop y ∙ emloop g
+          transp-emloop=∙-emloop = transp-cst=idf (emloop g) (emloop y)
+
+          ∙-emloop=emloop-comp : emloop' G y ∙ emloop g == emloop (G.comp y g)
+          ∙-emloop=emloop-comp = ! (emloop-comp y g)
+
+          emloop-comp=emloop-transp : emloop' G (G.comp y g) == emloop-transp (emloop' G g) y
+          emloop-comp=emloop-transp = ap emloop (! (↓-Codes-loop-transp g y))
+
+          transp-emloop=emloop-transp : transp-emloop (emloop g) y == emloop-transp (emloop g) y
+          transp-emloop=emloop-transp = transp-emloop=∙-emloop ∙ ∙-emloop=emloop-comp ∙ emloop-comp=emloop-transp
```

Comment:
(This is very minor!) Personally I think `...-p=q-...` would be confusing because it looks like `p=q` is more closely related than the other parts.